server/src/route/rule_based.rs

```diff
@@ -1,27 +1,40 @@
 // Copyright 2022 CeresDB Project Authors. Licensed under Apache-2.0.
 
+//! A router based on rules.
+
 use std::{
     collections::HashMap,
     hash::{Hash, Hasher},
-    sync::Arc,
 };
 
 use ceresdbproto_deps::ceresdbproto::storage::{Endpoint, Route, RouteRequest};
+use cluster::{Node, SchemaConfig};
 use log::info;
-use meta_client::{MetaClient, ShardId};
+use meta_client::types::ShardId;
 use serde_derive::Deserialize;
 use twox_hash::XxHash64;
 
-use crate::error::{ErrNoCause, Result, StatusCode};
+use crate::{
+    error::{ErrNoCause, Result, StatusCode},
+    route::Router,
+};
 
 /// Hash seed to build hasher. Modify the seed will result in different route
 /// result!
 const HASH_SEED: u64 = 0;
 
-pub type RouterRef = Arc<dyn Router + Sync + Send>;
+pub type ShardViews = HashMap<ShardId, ShardView>;
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct ShardView {
+    pub shard_id: ShardId,
+    pub node: Node,
+}
```

Comment:
This looks duplicated, other place defines `ShardView` too

Reply:
They are duplicated in name rather than content, so I will keep this intact and modify the `ShardView` in the cluster crate to `NodeShard`:
```rust
pub struct NodeShard {
    pub shard: ShardInfo,
    pub node: Node,
}
```

Reply:
```rust
pub struct ShardInfo {
    pub shard_id: ShardId,
    pub role: ShardRole,
}

pub enum ShardRole {
    LEADER,
    FOLLOWER,
}
```

But their contents are not that different I think 🧐 ?

Reply:
It's true, but `rule_based.rs` needs such `ShardView` without `ShardRole` for configurations and `cluster` crate needs such `NodeShard` with `ShardRole` to describe the cluster topology. So the two similar but different structs have to exist at the same time.

Reply:
I have moved this `ShardView` to server/config.rs as just a config struct. And in the rule_based.rs, only `Node` is necessary.